http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/DataprocInterpreter.scala

```diff
@@ -357,18 +357,41 @@ class DataprocInterpreter[F[_]: Parallel](
         )
         metadata <- getShutdownScript(params.runtimeAndRuntimeConfig, false)
         _ <- params.masterInstance.traverse { instance =>
-          googleComputeService
-            .addInstanceMetadata(instance.key.project, instance.key.zone, instance.key.name, metadata)
+          for {
+            opFutureAttempt <- googleComputeService
+              .addInstanceMetadata(instance.key.project, instance.key.zone, instance.key.name, metadata)
+              .attempt
+            _ <- opFutureAttempt match {
+              case Left(e) if e.getMessage.contains("Instance not found") =>
```

Comment:
We might want to log something here in addition to no-op

Reply:
I think from library perspective, if a VM doesn't exist, and client tries to update metadata, then it should error out. Do nothing might not be applicable to all cases